RecoTracker/MkFitCMS/standalone/MkStandaloneSeqs.cc

```diff
@@ -0,0 +1,1083 @@
+          const auto idx = simtrack.getHitIdx(ihit);
+
+          if (lyr < 0 || idx < 0)
+            continue;  // standard check
+
+          if (!seedHitIDMap.count(lyr))
+            continue;  // ensure seed hit map has at least one entry for this layer
+          if (!seedHitIDMap.at(lyr).count(idx))
+            continue;  // ensure seed hit map has at least one entry for this idx
+
+          for (const auto label : seedHitIDMap.at(lyr).at(idx)) {
+            const auto &seedLayers = labelSeedHitsMap[label];
+            if (std::find(seedLayers.begin(), seedLayers.end(), lyr) != seedLayers.end())  //seed check moved here
+              seedIDMap[label].emplace(lyr);
+          }
+        }
+
+        // now see if one of the seedIDs matched has at least 4 hits!
+        bool isSimSeed = false;
+        for (const auto &seedIDpair : seedIDMap) {
+          if ((int)seedIDpair.second.size() == labelNHitsMap[seedIDpair.first]) {
+            isSimSeed = true;
+            if (Config::mtvRequireSeeds)
+              simtrack.setAlgoint(labelAlgoMap[seedIDpair.first]);
+            if (Config::mtvRequireSeeds)
+              event->simTracksExtra_[count].addAlgo(labelAlgoMap[seedIDpair.first]);
+            //break;
+          }
+        }
+        if (Config::mtvLikeValidation) {
+          // Apply MTV selection criteria and then return
+          if (simtrack.prodType() != Track::ProdType::Signal || simtrack.charge() == 0 || simtrack.posR() > 2.5 ||
+              std::abs(simtrack.z()) > 30 || std::abs(simtrack.momEta()) > 3.0)
+            simtrack.setNotFindable();
+          else if (Config::mtvRequireSeeds && !isSimSeed)
+            simtrack.setNotFindable();
+        } else {
+          // set findability based on bool isSimSeed
+          if (!isSimSeed)
+            simtrack.setNotFindable();
+        }
+        count++;
+      }
+    }
+
+    void prep_cmsswtracks(Event *event) { prep_reftracks(event, event->cmsswTracks_, event->cmsswTracksExtra_, true); }
+
+    void prep_reftracks(Event *event, TrackVec &tracks, TrackExtraVec &extras, const bool realigntracks) {
+      prep_tracks(event, tracks, extras, realigntracks);
+
+      // mark cmsswtracks as unfindable if too short
+      for (auto &track : tracks) {
+        const int nlyr = track.nUniqueLayers();
+        if (nlyr < Config::cmsSelMinLayers)
+          track.setNotFindable();
+      }
+    }
+
+    void prep_tracks(Event *event, TrackVec &tracks, TrackExtraVec &extras, const bool realigntracks) {
+      for (size_t i = 0; i < tracks.size(); i++) {
+        extras.emplace_back(tracks[i].label());
+      }
+      if (realigntracks)
+        event->validation_.alignTracks(tracks, extras, false);
+    }
+
+    void score_tracks(TrackVec &tracks) {
+      for (auto &track : tracks) {
+        track.setScore(getScoreCand(track));
+      }
+    }
+
+  }  // namespace StdSeq
+
+}  // namespace mkfit
+
+//==============================
```

Comment:
I wish I knew :) This is was slightly aged standalone code that was rushed out of core mkfit classes (MkBuilder and MkFitter) when this PR was being prepared.
I have moved it out to another repo (trackreco/mkFit-external) as even if the code comes back it will have to be significantly changed.

Addressed in Round 5 follow-up PR trackreco/cmssw#80 commit: https://github.com/trackreco/cmssw/pull/80/commits/ccfaecfc8bee83cbd075c0ff30d2eef9210a33d8